src/Compilers/CSharp/Portable/Symbols/Metadata/PE/DynamicTypeDecoder.cs

```diff
@@ -338,6 +341,78 @@ private PointerTypeSymbol TransformPointerType(PointerTypeSymbol pointerType)
                 new PointerTypeSymbol(pointerType.PointedAtTypeWithAnnotations.WithTypeAndModifiers(transformedPointedAtType, pointerType.PointedAtTypeWithAnnotations.CustomModifiers));
         }
 
+#nullable enable
+        private FunctionPointerTypeSymbol? TransformFunctionPointerType(FunctionPointerTypeSymbol type)
+        {
+            var flag = ConsumeFlag();
+            Debug.Assert(!flag);
+
+            var sig = type.Signature;
+
+            var transformedReturn = handle(ref this, sig.RefKind, sig.RefCustomModifiers, sig.ReturnTypeWithAnnotations);
+            if (transformedReturn is null)
+            {
+                return null;
+            }
+
+            var transformedParameters = ImmutableArray<TypeWithAnnotations>.Empty;
+            var paramsTransformed = false;
+            if (sig.ParameterCount > 0)
+            {
+                var paramsBuilder = ArrayBuilder<TypeWithAnnotations>.GetInstance(sig.ParameterCount);
+                try
+                {
+                    foreach (var param in sig.Parameters)
+                    {
+                        var transformedParamType = handle(ref this, param.RefKind, param.RefCustomModifiers, param.TypeWithAnnotations);
+                        if (transformedParamType is null)
+                        {
+                            return null;
+                        }
+
+                        if (param.Type.Equals(transformedParamType, TypeCompareKind.ConsiderEverything))
+                        {
+                            paramsBuilder.Add(param.TypeWithAnnotations);
+                        }
+                        else
+                        {
+                            paramsBuilder.Add(param.TypeWithAnnotations.WithType(transformedParamType));
+                            paramsTransformed = true;
+                        }
+                    }
+
+                    transformedParameters = paramsTransformed ? paramsBuilder.ToImmutable() : sig.ParameterTypesWithAnnotations;
+                }
+                finally
+                {
+                    paramsBuilder.Free();
+                }
+            }
+
+            if (paramsTransformed || !sig.ReturnType.Equals(transformedReturn, TypeCompareKind.ConsiderEverything))
+            {
+                return type.SubstituteTypeSymbol(sig.ReturnTypeWithAnnotations.WithType(transformedReturn), transformedParameters,
```

Comment:
>sig.ReturnTypeWithAnnotations.WithType(transformedReturn) [](start = 49, length = 57)

t looks like this could be unnecessary if the type didn't change.